### PR TITLE
Create data model

### PR DESCRIPTION
**Pull request by liondon**
There are a few things that's different from our previous decision:
1. I used postgreSQL, following the settings in `lab-flask-tdd` repo. Because even though I can successfully run a MySQL container and use `docker exec -it mysql mysql -uroot -ppassword` to connect into it, I kept getting `CRITICAL in __init__: (pymysql.err.OperationalError) (1130, "172.17.0.1' is not allowed to connect to this MySQL server")`  while trying to run the flask server and connect to it with SQLAlchemy. (I've spent hours trying to fix this, but none of the solutions I found online can solve it.)

2. I added a dummy `Product` model for the `promotion_products` table, following the suggestion in [Flask documentation](https://flask-sqlalchemy.palletsprojects.com/en/2.x/models/). I'm not very sure about this and will check this with Professor.

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+"""
+Global Configuration for Application
+"""
+import os
+
+# Get configuration from environment
+# DATABASE_URI = os.getenv("DATABASE_URI", "sqlite:///../development.db")
```
Comment (nspektor) [inline config.py:7]: should we just delete this line?

File: service/models.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+- id: (int) primary key
+- title: (str) the name of the promotion
+- description: (str) string/text
+- promo_code: (str) the promo_code associated with this promotion 
+- promo_type: (str or enum) [BOGO | DISCOUNT | FIXED]
+- amount: (int) the amount of the promotion base on promo_type
+- start_date: (date) the starting date
+- end_date: (date) the ending date
+-----------
+promotion_products - The relationship between promotion and product
```
Comment (nspektor) [inline service/models.py:18]: Question: so for a site wide promotion, would we have to have a promotion_product entry for every product? Maybe we should instead have a field on the promotion table indicating if it is site wide? @liondon

File: service/service.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+Paths:
+------
+GET / - Returns a welcome message and 200 code
+"""
+import os
+import sys
+import logging
+from flask import Flask, jsonify, request, url_for, make_response, abort
+from flask_api import status  # HTTP Status Codes
+# from werkzeug.exceptions import NotFound
```
Comment (nspektor) [inline service/service.py:13]: why is this commented out? do we not need it anymore?

**Review by nspektor — CHANGES_REQUESTED** [review]

File: config.py (hunk 1 above)
Comment (liondon) [reply config.py:7]: yeah, surely we could do that!

File: service/models.py (hunk 2 above)
Comment (liondon) [reply service/models.py:18]: oh, yeah, you're right. I thought about this for one second but then forgot it :P

File: service/service.py (hunk 3 above)
Comment (liondon) [reply service/service.py:13]: I commented it out cause I didn't use it in my codes, but we might need it afterwards.

**Review by nspektor — APPROVED** [review]

**Pull request merged**